src/Views/AppInfoView.vala

```diff
@@ -519,8 +524,35 @@ namespace AppCenter.Views {
                     });
                 });
 
-                foreach (var url in urls) {
-                    load_screenshot (url);
+                string?[] screenshot_files = new string?[urls.length ()];
+                int[] results = new int[urls.length ()];
+                int completed = 0;
+
+                // Fetch each screenshot in parallel.
+                for (int i = 0; i < urls.length (); i++) {
+                    string url = urls.nth_data (i);
+                    string? file = null;
+                    int index = i;
+
+                    cache.fetch.begin (url, (obj, res) => {
+                        results[index] = cache.fetch.end (res, out file);
+                        screenshot_files[index] = file;
+                        completed++;
+                    });
+                }
+
+                cache.maintain ();
```

Comment:
Maybe let's just do the maintain once when you either open or close the app instead of going though all the cache every time?

Reply:
The time to run this is within the `~1ms` range.